cmd/plasmacli/config/config.go

```diff
@@ -3,66 +3,121 @@ package config
 import (
 	"bytes"
 	"fmt"
+	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/spf13/viper"
 	cmn "github.com/tendermint/tendermint/libs/common"
 	"os"
 	"path/filepath"
+	"strings"
 	"text/template"
 )
 
 const defaultConfigTemplate = `# This is a TOML config file.
 # For more information, see https://github.com/toml-lang/toml
 
-##### ethereum config options #####
+##### ethereum configuration #####
+
 # Ethereum plasma contract address
-ethereum_plasma_contract_address = "{{ .EthPlasmaContractAddr }}"
+ethereum_contract_address = "{{ .EthPlasmaContractAddr }}"
 
 # Node URL for eth client
 ethereum_nodeurl = "{{ .EthNodeURL }}"
 
-# Number of Ethereum blocks until a transaction is considered final
-ethereum_finality = "{{ .EthBlockFinality }}"`
+
+##### plasamd configuration #####
+
+# Node URL for plasmad
+node = "{{ .PlasmadNodeURL }}"
+
+# Trust the connected plasmad node (don't verify proofs for responses)
+trust_node = {{ .PlasmadTrustNode }}
+
+# Chain identifier. Must be set if trust-node == false
+chain_id = "{{ .PlasmadChainID }}"`
 
 // Must match the above defaultConfigTemplate
-type PlasmaConfig struct {
-	EthPlasmaContractAddr string `mapstructure:"ethereum_plasma_contract_address"`
+type Config struct {
+	// Ethereum config
+	EthPlasmaContractAddr string `mapstructure:"ethereum_contract_address"`
 	EthNodeURL            string `mapstructure:"ethereum_nodeurl"`
-	EthBlockFinality      string `mapstructure:"ethereum_finality"`
+
+	// Plasmad config
+	PlasmadNodeURL   string `mapstructure:"node"`
+	PlasmadTrustNode bool   `mapstructure:"trust_node"`
+	PlasmadChainID   string `mapstructure:"chain_id"`
 }
 
 var configTemplate *template.Template
 
 func init() {
 	var err error
-	tmpl := template.New("plasmaConfigFileTemplate")
+	tmpl := template.New("configFileTemplate")
 	if configTemplate, err = tmpl.Parse(defaultConfigTemplate); err != nil {
 		panic(err)
 	}
+
+	// ensure that none of the cosmos-cli flags have changed
+	var changed bool
```

Comment:
Why is this section necessary? Seems like its just meant to catch a future breaking change where the flag names change. If this happens I imagine it'll be a compilation error. Furthermore, I don't think these flags will ever change, or it'll be in an update doc if they did

Reply:
I can remove this. It was just to capture a future breaking change. Where would the compilation failure occur? If the flags name change, it would be silent error

If we're watching sdk changes, it should be fine though

Reply:
Yea I think we will notice via watching name changes in upgrades. Also if they changed the value of the flag they would probably change the variable name to match it

Reply:
@AdityaSripal thoughts?

Reply:
i'll remove it. We should just watch the changelog and do testing on each upgrade